resources/spotlight.rb

```diff
@@ -34,9 +34,10 @@ def desired_spotlight_state
 action :set do
   volume = MetadataUtil.new(target_volume)
 
-  execute 'Enable Spotlight server' do
-    command volume.toggle_spotlight_server(true)
-    only_if { volume.server_disabled? }
+  macosx_service 'spotlight server' do
```

Comment:
you should be able to just make this service. Chef does the right thing and wires up the provider under the hood